### ISSUE TITLE
Possible data race in `StrokeHandler`, `BaseStrokeHandler`, `SplineHandler` (possible others)

### ISSUE THREAD

**Issue by bhennion** [xournalpp/xournalpp#3579]
In each of the classes `StrokeHandler`, `BaseStrokeHandler` and `SplineHandler`, two parallel accesses to a stroke might trigger SegFaults:

- `draw()` reads to the stroke's point vector
- `onMotionNotifyEvent()` can push new points to the stroke's point vector.

Those functions are not called by the same thread (you can test this with `g_thread_self`).
A writing operation in `onMotionNotifyEvent()` could then trggier a `std::vector` reallocation, thus invalidating iterators used in `draw()`. This could lead to SegFaults

**Comment by Febbe** [comment]
Good catch

**Comment by bhennion** [comment]
I think I made a mistake when I reported this. This sentence is just plainly wrong:

> Those functions are not called by the same thread (you can test this with `g_thread_self`).

My assumption is that back then, I tested with the eraser tool (instead of with one of the InputHandlers).

GTK applications only have one main loop, and there is no drawing thread. The only other threads there are are those we create as jobs.

(Note that in the case of the eraser, the drawing operations are carried out in a RenderJob, so in a different thread. This is not the case for StrokeHandler and co.)
